libs/ui/src/button.tsx

```diff
@@ -294,9 +294,18 @@ function disabledStyles(p: ThemedStyledButtonProps): CSSObject {
     };
   }
 
+  const backgroundColor = isInverse(color)
+    ? colors.foreground
+    : colors.background;
+  const foregroundColor = isInverse(color)
+    ? colors.background
+    : colors.foreground;
+
   return {
+    backgroundColor,
     borderStyle: 'dashed',
-    borderColor: 'currentColor',
+    color: foregroundColor,
+    borderColor: foregroundColor,
```

Comment:
Thanks for calling this out. I think you're right that it doesn't look disabled enough when it maintains the existing background color.

I made this change because of components like `SegmentedButton` and `RadioGroup` that need to show which option was selected even when disabled. But I think we don't use those components in their disabled state very often (if at all) currently in touch screen mode, so I think it's more important to prioritize regular buttons looking disabled.

That being said, I do think that leaves a hole in our implementation of `SegmentedButton`/`RadioGroup` in touch mode for the time being.

Reply:
Ah, that makes sense! I've always thought of `role=option` elements to have different semantics from `button`s, so in this case, I'd propose that those components override the default styling to suit their semantics for the disabled state (in the same way we map their `selected` states to a `primary` button variant to begin with). Wouldn't want the `option` components to influence `Button` styling, IMO

Will take a look at those elements and see if it's low-lift enough to update them as part of this PR.

Reply:
Yeah I think that makes sense!